wallet/src/main/scala/org/bitcoins/wallet/internal/AddressHandling.scala

```diff
@@ -450,7 +450,7 @@ private[wallet] trait AddressHandling extends WalletLogger {
     * With this background thread, we poll the [[addressRequestQueue]] seeing if there
     * are any elements in it, if there are, we process them and complete the Promise in the queue.
     */
-  lazy val walletThread = new Thread(AddressQueueRunnable)
+  lazy val addressQueueThread = new Thread(AddressQueueRunnable)
```

Comment:
Didn't we talk about getting rid of these and just using the scheduler?

Reply:
woops, added

Reply:
So this still isn't solving the problem (you still allocate a `new Thread(AddressQueueRunnable)`)

I tried fixing this for you, but it appears we have design/architecture decisions that prevent us from doing this cleanly. Every _instance_ of a wallet class has it's own scheduler, which means we need to ensure that `wallet.start()` is called, else the scheduler doesn't have the tasks scheduled on it. 

I tried to make these changes with `AddressQueueRunnable` last night/this morning, but it is tough to do since we have a lot of methods that return a new `Wallet` instance (which is the right thing to do!, we want to maintain referential transparency!) 

When a new wallet instance is returned, we need to make sure that instance has had the `.start()` method called on it. If it doesn't, we won't have the re-broadcast transaction task scheduled on the scheduler, and we don't have the address queue runnable scheduled either. 

If the `AddressQueueRunnable` isn't scheduled, it is super obvious things aren't working as literally you won't be able to fetch a bitcoin address, but the re-broadcast logic is much more subtle since it doesn't occur that often. 


I attempted to hack together something working for the `AddressQueueRunnable`, but am giving up for now as it is getting to be too time consuming and we have higher priorities

https://github.com/Christewart/bitcoin-s-core/tree/2021-03-23-rm-addressthread-queue